pkg/reboot/calculator.go

```diff
@@ -23,102 +23,100 @@ const (
-	log                                                                     logr.Logger
-	k8sClient                                                               client.Client
-	highestCalculatedBatchNumber                                            int
-	isAgent                                                                 bool
+var _ RebootDurationCalculator = &rebootDurationCalculator{}
+
+type rebootDurationCalculator struct {
+	// storing the config here when reconciling it increases resilience in case of issues during remediation
+	snrConfig *v1alpha1.SelfNodeRemediationConfig
 }
 
-func NewAgentSafeTimeCalculator(k8sClient client.Client, wd watchdog.Watchdog, maxErrorThreshold int, apiCheckInterval, apiServerTimeout, peerDialTimeout, peerRequestTimeout, timeToAssumeNodeRebooted time.Duration) SafeTimeCalculator {
-	return &safeTimeCalculator{
-		wd:                       wd,
-		maxErrorThreshold:        maxErrorThreshold,
-		apiCheckInterval:         apiCheckInterval,
-		apiServerTimeout:         apiServerTimeout,
-		peerDialTimeout:          peerDialTimeout,
-		peerRequestTimeout:       peerRequestTimeout,
-		timeToAssumeNodeRebooted: timeToAssumeNodeRebooted,
-		k8sClient:                k8sClient,
-		isAgent:                  true,
-		log:                      ctrl.Log.WithName("safe-time-calculator"),
-	}
+func NewRebootDurationCalculator() RebootDurationCalculator {
+	return &rebootDurationCalculator{}
 }
 
-func NewManagerSafeTimeCalculator(k8sClient client.Client, timeToAssumeNodeRebooted time.Duration) SafeTimeCalculator {
-	return &safeTimeCalculator{
-		timeToAssumeNodeRebooted: timeToAssumeNodeRebooted,
-		k8sClient:                k8sClient,
-		isAgent:                  false,
-		log:                      ctrl.Log.WithName("safe-time-calculator"),
-	}
+func (r *rebootDurationCalculator) SetConfig(config *v1alpha1.SelfNodeRemediationConfig) {
+	r.snrConfig = config
 }
 
-func (s *safeTimeCalculator) GetTimeToAssumeNodeRebooted() time.Duration {
-	if !s.isAgent {
-		return s.timeToAssumeNodeRebooted
+// TODO add unit test!
+func (r *rebootDurationCalculator) GetRebootDuration(k8sClient client.Client, ctx context.Context, node *v1.Node, operatorNs string) (time.Duration, error) {
+
+	log := ctrl.Log.WithName("rebootDurationCalculator")
+
+	var config *v1alpha1.SelfNodeRemediationConfig
+	if r.snrConfig != nil {
+		config = r.snrConfig
+	} else {
+		// just in case we reconcile a SNR CR sooner than the SNRConfig CR
+		var err error
+		if config, err = r.getConfig(k8sClient, ctx, operatorNs); err != nil {
+			return 0, errors.Wrap(err, "failed to get SelfNodeRemediationConfig")
+		}
 	}
 
-	if s.timeToAssumeNodeRebooted < s.minTimeToAssumeNodeRebooted {
-		return s.minTimeToAssumeNodeRebooted
+	watchdogTimeout := utils.GetWatchdogTimeout(node)
+	minimumCalculatedRebootDuration, err := r.calculateMinimumRebootDuration(k8sClient, ctx, config, watchdogTimeout)
```

Comment:
yes and no: in case the config wasn't set yet, the config which was got within this method, wasn't set to the field (which didn't make that much sense...).

However, after removing that part as suggested in your other comment, there only is yes to this 😉 Done